components/camel-jta/src/main/java/org/apache/camel/jta/JtaTransactionErrorHandler.java

```diff
@@ -64,4 +69,16 @@ public ErrorHandler clone(Processor output) {
         return answer;
     }
 
+    @Override
+    public boolean process(final Exchange exchange, final AsyncCallback callback) {
+        if (!exchange.isTransacted()) {
+            LOG.debug("mark " + exchange + " transacted");
+            exchange.getUnitOfWork().beginTransactedBy("jta");
+            super.process(exchange, callback);
+            exchange.getUnitOfWork().endTransactedBy("jta");
+        } else {
+            super.process(exchange, callback);
+        }
+        return false;
```

Comment:
This may be an existing "issue" as the returned value should be the output from super.process and not hardcoded to false